### PR TITLE
set equinox configuration app args

### PR DESCRIPTION
**Pull request by gamerson**

### CODE REVIEW

File: org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java [hunk 1]
```diff
@@ -69,6 +73,38 @@ private static Filter createFilter() {
...
+							equinoxConfig.setAllArgs(parameters);
+							equinoxConfig.setAppArgs(parameters);
+						}
+					}
+
+					log.fine("configured program arguments " + parameters);
+				}
+			} else {
+				log.log(Level.SEVERE,
+					"service aQute.launcher.Launcher with props launcher.arguments could not be retireved");
```
Comment (rotty3000) [inline org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java:101]: Typo in `retrieved`

File: org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java [hunk 2]
```diff
@@ -69,6 +75,39 @@ private static Filter createFilter() {
...
+					if (srvcRef != null) {
+						EnvironmentInfo service = context.getService(srvcRef);
+						if (service instanceof EquinoxConfiguration) {
+							EquinoxConfiguration equinoxConfig = (EquinoxConfiguration) service;
+							equinoxConfig.setAllArgs(args);
+							equinoxConfig.setAppArgs(args);
+						}
+					}
+
+					log.fine("configured program arguments " + Arrays.stream(args)
```
Comment (bjhargrave) [inline org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java:99]: I think `Arrays.toString(args)` is probably fine and simpler.

Why is this log `log.fine(` when the other logs are like `log.log(Level.`? It seems a bit inconsistent.

File: org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java [hunk 3]
```diff
@@ -69,6 +75,39 @@ private static Filter createFilter() {
...
+				"(launcher.arguments=*)");
+			if (serviceReferences != null && serviceReferences.length > 0) {
+				Object property = serviceReferences[0].getProperty("launcher.arguments");
+
+				if (property instanceof String[]) {
+					args = (String[]) property;
+
+					ServiceReference<EnvironmentInfo> srvcRef = context.getServiceReference(EnvironmentInfo.class);
+					if (srvcRef != null) {
+						EnvironmentInfo service = context.getService(srvcRef);
```
Comment (bjhargrave) [inline org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java:91]: What about `unget`ing the service?

File: org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java [hunk 4]
```diff
@@ -69,6 +75,39 @@ private static Filter createFilter() {
...
+						if (service instanceof EquinoxConfiguration) {
+							EquinoxConfiguration equinoxConfig = (EquinoxConfiguration) service;
+							equinoxConfig.setAllArgs(args);
+							equinoxConfig.setAppArgs(args);
```
Comment (bjhargrave) [inline org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java:95]: Why are the command line args (`setAllArgs`) and the non-framework args (`setAppArgs`) set to the same value?

Comment (gamerson) [reply org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java:95]: AppArgs is all I need so I'll remove the other.

File: org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java (hunk 3 above)
Comment (gamerson) [reply org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java:91]: I'll add the unget

File: org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java (hunk 2 above)
Comment (gamerson) [reply org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java:99]: I'll clean those up.

File: org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java [hunk 5]
```diff
@@ -69,8 +75,40 @@ private static Filter createFilter() {
...
+					"service aQute.launcher.Launcher with props launcher.arguments could not be retrieved");
+			}
+		} catch (Exception e) {
+			log.log(Level.SEVERE, "command line parameters could not be configured.");
+		}
+	}
+
 	@Override
 	public void removedService(ServiceReference<Object> reference, ServiceRegistration<ApplicationLauncher> service) {
+		context.ungetService(envInfo);
```
Comment (bjhargrave) [inline org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java:111]: `envInfo` could be null here, no?

Comment (gamerson) [reply org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java:111]: ah yes

**Pull request merged**